flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -651,7 +651,7 @@ boolean isSerializingTimestamps() {
 	 * Gets the name of the task, in the form "taskname (2/5)".
 	 * @return The name of the task.
 	 */
-	public String getName() {
+	public final String getName() {
```

Comment:
I guess the motivation for defining `final` for related parent methods is from constructing `SubtaskCheckpointCoordinatorImpl` in `StreamTask` constructor. If so, we need also define final for `AbstractInvokable#getEnvironment` because it is also accessed while constructing `SubtaskCheckpointCoordinatorImpl`.

Actually we can avoid using getter while constructing `SubtaskCheckpointCoordinatorImpl` to use specific arguments instead. Anyway i think it is meaningful to define `final` for some methods if we confirm they should not be override by subclasses. But the requirement should not be from constructing `SubtaskCheckpointCoordinatorImpl`, and we should review all the methods in `AbstractInvokable` and `StreamTask` to make this thing completely.

Reply:
Right, the motivation is the use of this method in constructor.
Thanks for pointing out about `getEnvironment` (it was modified made `final` too but then reverted accidentally while fixing tests).

I did consider the alternative of using specific values in constructor, but then we can get inconsistency between values used in constructor with ones in getters. So using getters here and declaring them `final` I think is a lesser evil.

Regarding the contract of `AbstractInvokable` and `StreamTask`, I don't think these methods are extension points (as opposed to `processInput` for example), and therefore shouldn't be overridden.

Reply:
Yes, i agree with your consideration for `inconsistency between values used in constructor with ones in getters`.

Regarding the `AbstractInvokable` and `StreamTask`, you misunderstood my previous comment. I mean that there might still have other methods inside parent `AbstractInvokable` and `StreamTask`,  which should not be overridden by subclasses, so it should also be defined as final as you already did in this PR. Now we only consider the requirements from `SubtaskCheckpointCoordinatorImpl` to handle the related methods. But from a more general motivation, we might need to go through all the necessary methods to make them final to make this story complete. But it is up to you whether to touch more things in this PR.